RHIDP-8641: Building and Deploying Serverless Workflows (#1416)
redhat-developer/red-hat-developers-documentation-rhdh: modules/orchestrator/proc-deploying-workflows-on-a-cluster.adoc

:_mod-docs-content-type: PROCEDURE

[id="proc-deploying-workflows-on-a-cluster.adoc_{context}"]
= Deploying workflows on a cluster

> You can deploy the workflow on a cluster, since the image has been pushed to the image registry and the deployment manifests have been made available.

Reviewer: ```suggestion
You can deploy the workflow on a cluster, since the image is pushed to the image registry and the deployment manifests are available.
```
(I'm not sure if this grammatical change is technically accurate. Please ignore if this does not apply)

.Prerequisites

* You have an {ocp-short} cluster with the following versions of components installed:

> *   {product} ({product-very-short}) `v1.5`

Reviewer 2: I would align versions with 1.7 - so plugins v1.7.1 and serverless and serverless logic should be 1.36

*   Orchestrator plugins `v1.4` or `v1.5`
*   OpenShift Serverless `v1.35`
*   OpenShift Serverless Logic `v1.35`

For further instructions on installing these components, see the {orchestrator-book-link}#con-orchestrator-plugin-components[Orchestrator plugin components on {ocp-short}].

> * You must apply the workflow manifests in a namespace that contains a `SonataflowPlatform` custom resource (CR), which manages the link:https://sonataflow.org/serverlessworkflow/latest/cloud/operator/supporting-services.html[supporting services]. By default, OpenShift Serverless Logic services are installed in the `sonataflow-infra` namespace.

Reviewer 2: What do you think about using https://docs.redhat.com/en/documentation/red_hat_openshift_serverless/1.36/html-single/serverless_logic/index#serverless-logic-configuring-workflow-services

> * You must apply the workflow manifests in a namespace that contains a `SonataflowPlatform` custom resource (CR), which manages the link:https://sonataflow.org/serverlessworkflow/latest/cloud/operator/supporting-services.html[supporting services]. By default, OpenShift Serverless Logic services are installed in the `sonataflow-infra` namespace.

Reviewer 3: `By default, OpenShift Serverless Logic services are installed in the `sonataflow-infra` namespace.` - I am not sure this is still true? @pkliczewski

Reviewer 4: When I installed RHDH 1.7.1 on a fresh cluster, all OSL resources were installed in the `rhdh-operator` namespace which is the default
There is this doc https://github.com/redhat-developer/rhdh-operator/blob/main/docs/orchestrator.md#optional-enabling-workflow-in-a-different-namespace to use a different namespace

the Job Service and Data Index (and other OSL related resources) used to be installed in the sonataflow-infra be default (but not anymore)

...
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Failed to read configuration properties
----
+
The error indicates a missing property: `quarkus.openapi-generator.notifications.auth.BearerToken.bearer-token`.

. In such a case where the logs show the `ConfigurationException: Failed to read configuration properties` error or indicate a missing value, retrieve the ConfigMap as shown in the following example:
+
[source,yaml]
----
oc get -n sonataflow-infra configmaps basic-props -o yaml
----
+
The sample output looks as in the following example:
+
[source,yaml]
----
apiVersion: v1
data:
  application.properties: |
    # Backstage notifications service
    quarkus.rest-client.notifications.url=${BACKSTAGE_NOTIFICATIONS_URL}
    quarkus.openapi-generator.notifications.auth.BearerToken.bearer-token=${NOTIFICATIONS_BEARER_TOKEN}
...
----
+
Resolve the placeholders using values provided using a Secret.

. You must edit the corresponding Secret and provide appropriate base64-encoded values to resolve the placeholders in `application.properties` as shown in the following example:
+
[source,yaml]
----
kubectl edit secrets -n sonataflow-infra basic-secrets
----

> . Restart the workflow Pod for Secret changes to take effect in OpenShift Serverless Logic `v1.35`.

Reviewer 2: let's update the version to 1.36
